### DISCUSSION TITLE
General questions

### DISCUSSION THREAD

**Discussion by seidler2547** [Intermesh/groupoffice#1050] [Q&A]
Hi, I'm just starting to see if Group-Office could be my next Mail/Groupware server software and I have a couple of questions that I hope someone can answer. So far I'm quite impressed with what I see but there are some things that I'm wondering about.

1. Is there a vertical preview pane layout in emails? Having the list of emails on top and the preview pane underneath it? I couldn't find a setting and that's something that feels quite a normal feature to me.
2. Why can I search the "Full Message" in the simple search but not the advanced search? I would really like to be able to search through all mail messages just using a basic search field, like Outlook Web, like Gmail, like Zimbra and many others. I have installed solr and dovecot-solr, so performance should not be an issue.
3. When I connect an Android phone via ActiveSync, it works very well, but no subfolders are shown. Is this a limitation of ActiveSync or is there another reason?
4. Is there something like "domain admins" in Group-Office? Meaning giving permissions to create and manage users but limited to a pre-selected domain? I host email for a couple of NGOs here and of course I don't want them interfering with other domains but their own.
5. As a normal user, I can just add any alias to my account and send from it, even though that email doesn't exist. Is that normal? Can it be prevented?
6. Is there a way to show all mail folders without having to subscribe to them?

**Comment by mschering** [comment]
Hi,

Thanks for trying Group-Office!

Here are my answers:

1. Is there a vertical preview pane layout in emails? Having the list of emails on top and the preview pane underneath it? I couldn't find a setting and that's something that feels quite a normal feature to me.

Unfortunately not. We used to have this but as most screens were more wide and we wanted to make it responsive for mobile phones we dropped this view. Maybe we'll bring it back as we're going to rewrite the mail module soon.

3. Why can I search the "Full Message" in the simple search but not the advanced search? I would really like to be able to search through all mail messages just using a basic search field, like Outlook Web, like Gmail, like Zimbra and many others. I have installed solr and dovecot-solr, so performance should not be an issue.

It's actually there but it's called "contents".

4. When I connect an Android phone via ActiveSync, it works very well, but no subfolders are shown. Is this a limitation of ActiveSync or is there another reason?

Not sure, I see my subfolders on my iphone. Maybe you need to subscribe them?

5. Is there something like "domain admins" in Group-Office? Meaning giving permissions to create and manage users but limited to a pre-selected domain? I host email for a couple of NGOs here and of course I don't want them interfering with other domains but their own.

Yes, You can setup permissions per domain.

6. As a normal user, I can just add any alias to my account and send from it, even though that email doesn't exist. Is that normal? Can it be prevented?

This can be done with any mail client. You can prevent it by not allowing manage permissions to the module and account so they can just use it but can't edit the account.

7. Is there a way to show all mail folders without having to subscribe to them?

No there isn't.

Best regards,
Merijn

**Reply by seidler2547** [reply]
Thanks for the details responses. Vertical preview pane would be great.

My advanced search looks like this:
![image](https://github.com/Intermesh/groupoffice/assets/729588/c01d621a-39aa-480d-abcb-5c9e932f0f91)
Not seeing "Contents".

Outlook on Android has caught up on the folders, but they're all empty. But it might be an Outlook problem, I will try another client.

About the aliases - the bigger problem is that the postfix configuration allows sending from any alias, not only the ones set up in the frontend. I checked the configuration though and the cause was not immediately obvious to me.

**Reply by seidler2547** [reply]
Oh, and I saw now that I can restrict the domains, but that only allows to create mailboxes. Can I allow another user to create new users and modify existing users, but only within their own domain?

**Reply by mschering** [reply]
Sorry, translation mixup. Contents should be "body".

Yes, You can setup permissions per domain within the "E-mail domains" module so they can only edit within there own domain.

**Reply by awnz** [reply]
I couldn't get this to work without patching MessagesGrid.js and EmailClient.js to search all mail, anywhere, for any keyword.

This is the modifcation I made to have a "Search everywhere" option to suit our needs:

MessagesGrid.js
```
227a228,232
>               this.searchTypeButton.menu.insert(6, {
>                       value: 'everywhere',
>                       text:  t("Search everywhere", "email"),
>                       icon:  'email'
>               });

```

EmailClient.js
```
158a159,162
>                       } else if(search_type=='everywhere') {
>                               query = 'TEXT "' + GO.email.search_query + '"';
>                               searchIn = 'all';
>                               store.baseParams['searchIn'] = searchIn;
```

I've deliberately placed the MessagesGrid.js patch to only have an effect when `settings.config.email_allow_body_search==true`.

This is the result:
![image](https://github.com/Intermesh/groupoffice/assets/8421332/7c2f3d28-8fee-466b-b84d-326f8fe8b6f1)



After some more testing (with a larger message sample once I have one), would this be desirable for a PR?

(edits: better screenshot, more detail)

**Reply by mschering** [reply]
Doesn't this give the same results as "Full message"? Which IMAP server do you use?

**Reply by awnz** [reply]
Hi, no, "Full message" only searches the current folder:
![image](https://github.com/Intermesh/groupoffice/assets/8421332/685b1817-e011-462d-aeca-ad0745ce8fd0)

(although I just discovered a bug in my above patch, when you select "Search everywhere, `searchIn = 'all'` is set, but when something else is selected, it is not cleared)

**Reply by mschering** [reply]
But TEXT should also search the current folder only if I'm not mistaken. Only via advanced you can select search all folders.

**Reply by awnz** [reply]
Correct, until I did the patch 😉
But the "TEXT" option isn't available under the advanced search (others including BODY are, but not TEXT).
To search everything, you need `query="TEXT (searchstring)"` plus `searchIn = 'all'` to get all the folders - there was no way presented in the UI to get that combination that I could see.

edit: fixed unbalanced quote

**Reply by mschering** [reply]
Ah now I see :)
